docs/ext/commands/slash_commands.rst

```diff
@@ -454,3 +456,163 @@ Injections
+    async def add_5(inter: disnake.ApplicationCommandInteraction, num: int):
+        """
+        Adds 5 to a number. {{ADD_NUM}}
+
+        Parameters
+        ----------
+        num: Some number {{ COOL_NUMBER }}
+        """
+        await inter.response.send_message(f"{num} + 5 = {num + 5}")
+
+The keys ``{{XYZ}}`` are automatically extracted from the docstrings,
+and used for looking up names ``XYZ_NAME`` and descriptions ``XYZ_DESCRIPTION``.
+Note that whitespace is ignored, and the positioning inside the line doesn't matter.
+
+For instance, to add German localizations, create a ``locale/de.json`` file;
+the directory name/path can be changed arbitrarily, ``locale`` is just the one used here:
+
+.. code-block:: json
+
+    {
+        "ADD_NUM_NAME": "addiere_5",
+        "ADD_NUM_DESCRIPTION": "Addiere 5 zu einer anderen Zahl.",
+        "COOL_NUMBER_NAME": "zahl",
+        "COOL_NUMBER_DESCRIPTION": "Eine Zahl",
+    }
+
+To load a directory or file containing localizations, use :func:`bot.i18n.load(path) <i18n.LocalizationStore.load>`:
+
+.. code-block:: python3
+
+    ...
+    bot.i18n.load("locale/")  # loads all files in the "locale/" directory
+    bot.run(...)
+
+.. note::
+    If :attr:`Bot.reload <ext.commands.Bot.reload>` is ``True``, all currently loaded localization files
+    are reloaded when an extension gets automatically reloaded.
+
+
+.. _localizations_strict:
+
+Strict Localization
++++++++++++++++++++
+
+By default, missing keys that couldn't be found are silently ignored.
+To instead raise an exception when a key is missing, pass the ``strict_localization=True`` parameter to the client/bot constructor
+(see :attr:`Bot.strict_localization <ext.commands.Bot.strict_localization>`).
+
+
+Customization
++++++++++++++
+
+If you want more customization or low-level control over localizations, you can specify arbitrary keys for the commands/options directly.
+:class:`.Localized` takes the non-localized string (optional depending on target type) to keep the ability of e.g.
+overwriting ``name`` in the command decorator, and either a ``key`` or ``data`` parameter.
+
+This would create the same command as the code above, though you're free to change the keys like ``ADD_NUM_DESCRIPTION`` however you want:
+
+.. code-block:: python3
+
+    @bot.slash_command(name=Localized("add_5", key="ADD_NUM_NAME"), description=Localized(key="ADD_NUM_DESCRIPTION"))
+    async def _add_5_slash(
+        inter: disnake.ApplicationCommandInteraction,
+        num: int = commands.Param(
+            name=Localized(key="COOL_NUMBER_NAME"),
+            description=Localized(key="COOL_NUMBER_DESCRIPTION")
+        ),
+    ):
+        """
+        Adds 5 to a number.
+
+        Parameters
+        ----------
+        num: Some number
+        """
+        await inter.response.send_message(f"{num} + 5 = {num + 5}")
+
+While not recommended, it is also possible to not use ``.json`` files at all and instead specify localizations directly in the code:
```

Comment:
...it is also possible to avoid using ``.json`` files at all...

Reply:
0c2c87bbe38a5ee7aeef673890a821d95185c5b7